### PR TITLE
Regular rebuild and release of images

### PR DESCRIPTION
**Pull request by fwilhe**
The purpose of this PR is to provide updated Docker images on a regular basis, both for latest as for releases. This will also get rid of the (in my experience) unreliable autobuilds on Docker hub.

The idea is to week-daily push new latest images, and once a week to an automated release. This way, the users are not forced to use outdated software versions when using the images.

Todo before merge:

- [x] Disable auto-builds on docker hub

### CODE REVIEW

File: .github/workflows/push-master.yml [hunk 1]
```diff
@@ -1,40 +1,16 @@
+name: Build images and push master commits or on schedule
```
Comment (stippi2) [inline .github/workflows/push-master.yml]: Sentence this compute not does.

File: .github/workflows/push-release.yml [hunk 2]
```diff
@@ -1,25 +1,13 @@
...
-        # Check for 'SEVERE' messages in Jenkins log (usually caused by plugin or configuration issues)
-        docker logs cx-jenkins-master-test 2> >(grep --quiet SEVERE)
-    - name: Push to Docker Hub
-      run: |
-        docker tag ppiper/jenkins-master:RC ppiper/jenkins-master:${GITHUB_REF##*/}
-        echo ${{ secrets.DOCKERHUB_PASSWORD }} | docker login -u ${{ secrets.DOCKERHUB_USER }} --password-stdin
-        docker push ppiper/jenkins-master:${GITHUB_REF##*/}
-        rm -f /home/runner/.docker/config.json
+    - name: Build, test and push
+      run: ./ci.sh ${GITHUB_REF##*/} ${{ secrets.DOCKERHUB_PASSWORD }} ${{ secrets.DOCKERHUB_USER }}
```
Comment (stippi2) [inline .github/workflows/push-release.yml]: `${GITHUB_REF##*/}` will resolve to "v*" in this case?

File: ci.sh [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+smoke_test() {
+    # Test Jenkins master image
+    docker run --name cx-jenkins-master-test --detach --publish 8080:8080 \
+        -v /var/run/docker.sock:/var/run/docker.sock \
+        ppiper/jenkins-master:${VERSION}-RC
+    # Wait up to 5 minutes for Jenkins to be up and fail if it does not return 200 after this time
+    docker exec cx-jenkins-master-test timeout 300 bash -c "until curl --fail http://localhost:8080/api/json; do sleep 5; done"
+    # Check for 'SEVERE' messages in Jenkins log (usually caused by plugin or configuration issues)
+    docker logs cx-jenkins-master-test 2> >(grep --quiet SEVERE)
```
Comment (stippi2) [inline ci.sh]: Are the SEVERE entries just piped to stderr, or does the presence of such entries actually have an effect? (I am not proficient in bash scripts...)

File: ci.sh [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+push_image() {
+    local TAG=$1
+
+    docker tag ${TAG}:${VERSION}-RC ${TAG}:${VERSION}
+
+    if [ "x${VERSION}" = xlatest ]; then
+        echo docker push ${TAG}:backup-of-latest
+    fi
+
+#fixme echos
```
Comment (stippi2) [inline ci.sh]: What exactly still needs to be done (with regard to "fixme")?

**Comment by fwilhe** [comment]
Tested it here in a separate repo https://github.com/fwilhe/act/actions

File: ci.sh (hunk 4 above)
Comment (fwilhe) [reply ci.sh]: remove the `echo` before `docker push`

File: ci.sh (hunk 3 above)
Comment (fwilhe) [reply ci.sh]: it will exit with a non-zero exit code if any findings are there

File: .github/workflows/push-release.yml (hunk 2 above)
Comment (fwilhe) [reply .github/workflows/push-release.yml]: yes

**Comment by stippi2** [comment]
Looks good, but I don't really feel qualified to approve.

**Pull request merged**